appcli/models/cli_context.py

```diff
@@ -38,31 +38,53 @@ class CliContext(NamedTuple):
     # derived data
     # ---------------------------------
 
-    key_file: Path
-    """ File containing key for encryption/decryption. """
+    def get_key_file(self) -> Path:
+        """Get the location of the key file for decrypting secrets
 
-    generated_configuration_dir: Path
-    """ Directory to store the generated configuration files to. """
+        Returns:
+            Path: location of the key file
+        """
+        return Path(self.configuration_dir, "key")
 
-    app_configuration_file: Path
-    """
-    Path to a YAML file containing variables which are applied to the
-    templates to generate the final configuration files.
-    """
+    def get_generated_configuration_dir(self) -> Path:
+        """Get the directory containing the generated configuration
 
-    templates_dir: Path
-    """
-    Directory containing jinja2 templates used to generate the final
-    configuration files.
-    """
+        Returns:
+            Path: directory of generated configuration
+        """
+        return self.configuration_dir.joinpath(".generated")
 
-    project_name: str
-    """ Project name for launching docker containers/networks. """
+    def get_app_configuration_file(self) -> Path:
+        """Get the location of the configuration file
+
+        Returns:
+            Path: location of the configuration file
+        """
+        return self.configuration_dir.joinpath(f"{self.app_name.lower()}.yml")
+
+    def get_templates_dir(self) -> Path:
+        """Get the directory of the raw configuration templates
+
+        Returns:
+            Path: directory of configuration templates
+        """
+        return self.configuration_dir.joinpath("templates")
+
+    def get_project_name(self) -> str:
+        """Get a unique name for the application and environment
+
+        Returns:
+            str: the project name
+        """
+        return f"{self.app_name}_{self.environment}"
 
     # ---------------------------------
     # cli build data
     # ---------------------------------
 
+    app_name: str
+    """ The application's name """
+
     app_version: str
     """ The application's version """
 
```

Comment:
move `cli build data` block up above the `derived data` block, better readability